### PR TITLE
create a diff for review J2M response

### PR DESCRIPTION
**Pull request by cqi1217**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/envoy/extensions/filters/http/json_to_metadata/v3/json_to_metadata.proto [hunk 1]
```diff
@@ -112,5 +112,8 @@ message JsonToMetadata {
   }
 
   // Rules to match json body of requests
-  MatchRules request_rules = 1 [(validate.rules).message = {required: true}];
+  MatchRules request_rules = 1 [(validate.rules).message = {required: false}];
```
Comment (JuniorHsu) [inline api/envoy/extensions/filters/http/json_to_metadata/v3/json_to_metadata.proto]: just remove the `(validate.rules).message` clauses here and there

File: source/extensions/filters/http/json_to_metadata/filter.cc [hunk 2]
```diff
@@ -100,6 +103,16 @@ Rules FilterConfig::generateRequestRules(
   return rules;
 }
 
+Rules FilterConfig::generateResponseRules(
+    const envoy::extensions::filters::http::json_to_metadata::v3::JsonToMetadata& proto_config)
```
Comment (JuniorHsu) [inline source/extensions/filters/http/json_to_metadata/filter.cc:107]: ideally we want to DRY here. e.g., 
`generateRules(const envoy::extensions::filters::http::json_to_metadata::v3::Rules&)` here 
and construct `request_rules_(generateRules(proto_config.response_rules().rules()))` in constructor

same thing with allow content type and allow empty type

File: test/extensions/filters/http/json_to_metadata/filter_test.cc [hunk 3]
```diff
@@ -67,12 +67,33 @@ class FilterTest : public testing::Test {
       preserve_existing_metadata_value: true
 )EOF";
 
+  const std::string response_config_yaml_ = R"EOF(
```
Comment (JuniorHsu) [inline test/extensions/filters/http/json_to_metadata/filter_test.cc]: we can have just one default yaml i think?

**Review by JuniorHsu — CHANGES_REQUESTED** [review]
`auto& filter_metadata = decoder_callbacks_->streamInfo().dynamicMetadata().filter_metadata();`

this needs change in `Filter::addMetadata` too

File: source/extensions/filters/http/json_to_metadata/filter.cc (hunk 2 above)
Comment (cqi1217) [reply source/extensions/filters/http/json_to_metadata/filter.cc:107]: > ideally we want to DRY here. e.g., `generateRules(const envoy::extensions::filters::http::json_to_metadata::v3::Rules&)` here and construct `request_rules_(generateRules(proto_config.response_rules().rules()))` in constructor
> 
> same thing with allow content type and allow empty type

might optimize this in the future when contribute to upstream

**Pull request merged**